feeder/src/vote.ts

```diff
@@ -294,34 +295,66 @@ interface VoteArgs {
   gasPrices: string
 }
 
-export async function vote(args: VoteArgs): Promise<void> {
-  const client = new LCDClient({
-    URL: args.lcdAddress,
+function buildLCDClientConfig(args: VoteArgs, lcdIndex: number): LCDClientConfig {
+  return {
+    URL: args.lcdAddress[lcdIndex],
     chainID: args.chainID,
     gasPrices: args.gasPrices,
-  })
+  }
+}
+
+export async function vote(args: VoteArgs): Promise<void> {
   const rawKey: RawKey = await initKey(args.keyPath, args.password)
   const valAddrs = args.validator || [rawKey.valAddress]
   const voterAddr = rawKey.accAddress
-  const wallet = new Wallet(client, rawKey)
+
+  const lcdRotate = {
+    client: new LCDClient(buildLCDClientConfig(args, 0)),
+    current: 0,
+    max: args.lcdAddress.length - 1,
+  }
 
   while (true) {
     const startTime = Date.now()
 
-    await processVote(client, wallet, args.source, valAddrs, voterAddr).catch((err) => {
+    await processVote(
+      lcdRotate.client,
+      lcdRotate.client.wallet(rawKey),
+      args.source,
+      valAddrs,
+      voterAddr
+    ).catch((err) => {
       if (err.isAxiosError && err.response) {
         console.error(err.message, err.response.data)
       } else {
         console.error(err.message)
       }
 
+      if (err.isAxiosError) {
+        console.info('vote: lcd client unavailable, rotating to next lcd client.')
+        rotateLCD(args, lcdRotate)
+      }
+
       resetPrevote()
     })
 
     await Bluebird.delay(Math.max(500, 500 - (Date.now() - startTime)))
   }
 }
 
+function rotateLCD(args: VoteArgs, lcdRotate: { client: LCDClient; current: number; max: number }) {
+  if (++lcdRotate.current > lcdRotate.max) {
```

Comment:
I think we need to destroy previous LCDClient .. before rotate it. Is there a function?

Reply:
How to destroy it?
Currently it overrides the previous client.